dispatches/models/fossil_case/properties/hitecsalt_properties.py

```diff
@@ -196,42 +197,11 @@ def initialize(blk, state_args=None,
             If hold_states is True, returns a dict containing flags for
             which states were fixed during initialization.
         """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="properties")
         # Fix state variables if not already fixed by the control volume block
         if state_vars_fixed is False:
             # Fix state variables if not already fixed
-            Fflag = {}
-            Tflag = {}
-            Pflag = {}
-            
-            for k in blk.keys():
-                if blk[k].flow_mass.fixed is True:
-                    Fflag[k] = True
-                else:
-                    Fflag[k] = False
-                    if state_args is None:
-                        blk[k].flow_mass.fix()
-                    else:
-                        blk[k].flow_mass.fix(state_args["flow_mass"])
-
-                if blk[k].pressure.fixed is True:
-                    Pflag[k] = True
-                else:
-                    Pflag[k] = False
-                    if state_args is None:
-                        blk[k].pressure.fix()
-                    else:
-                        blk[k].pressure.fix(state_args["pressure"])
-
-                if blk[k].temperature.fixed is True:
-                    Tflag[k] = True
-                else:
-                    Tflag[k] = False
-                    if state_args is None:
-                        blk[k].temperature.fix()
-                    else:
-                        blk[k].temperature.fix(state_args["temperature"])
-
-            flags = {"Fflag": Fflag, "Tflag": Tflag, "Pflag": Pflag}
+            flags = fix_state_vars(blk, state_args)
 
         else:
             # Check when the state vars are fixed already result in dof 0
```

Comment:
I would suggest that this check be run in all cases to confirm that the degrees of freedom are zero, and not only in the case where `state_vars_fixed` is `True`. It never hurts to be sure.